pkg/api/dataproxy.go

```diff
@@ -115,12 +118,28 @@ func ProxyDataSourceRequest(c *middleware.Context) {
 		}
 	}
 
+	outputToAuditLog(ds.Type, c)
+
 	proxy := NewReverseProxy(ds, proxyPath, targetUrl)
 	proxy.Transport, err = ds.GetHttpTransport()
 	if err != nil {
 		c.JsonApiErr(400, "Unable to load TLS certificate", err)
 		return
 	}
+
 	proxy.ServeHTTP(c.Resp, c.Req.Request)
 	c.Resp.Header().Del("Set-Cookie")
 }
+
+func outputToAuditLog(dataSourceType string, c *middleware.Context) {
+	auditLogger := log.New("data-proxy-audit", "userid", c.UserId, "orgid", c.OrgId, "username", c.Login)
+
+	var body string
+	if c.Req.Request.Body != nil {
+		buffer, _ := ioutil.ReadAll(c.Req.Request.Body)
+		c.Req.Request.Body = ioutil.NopCloser(bytes.NewBuffer(buffer))
+		body = string(buffer)
+	}
+
+	auditLogger.Info("Proxying incoming request", "datasource", dataSourceType, "uri", c.Req.RequestURI, "method", c.Req.Request.Method, "body", body)
```

Comment:
Ok, seems fair enough. I've followed the example of `router_logging` and added `data_proxy_logging` setting to the server settings and renamed things from audit -> data proxy in general. Agreed that a more fleshed out solution, separate log file, etc will be good. Personally, I don't even really like relying on info level here and expect some additional slowness, but having an auditable query stream is a big requirement for us in the near term. Thanks for quick replies!